Improve style

tagliala opened the pull request:
Contains changes (some are opinionated) for better integration with Bootstrap 4

### Add stylelint

- Same configuration as Boostrap
- Allows BEM-like selectors
- Some rules have been disabled ad-hoc

### Move layout to its own partial
Allow users to customize individual components or to use just one of them (e.g.,
"single" and "layout" when "multiple" is not needed)

### Improve single component
- Lint style
- Use $input-color to match Bootstrap
- Use $input-padding-x to match Bootstrap
- Use $input-placeholder-color to match Bootstrap

### Improve multiple component
- Lint style
- Use $input-padding-x to match Bootstrap
- Use $input-color to match Bootstrap

### Improve layout partial
- Lint style
- Use border-bottom-radius mixin
- Fix border radius on open containers
- Use color-yiq mixin to pick the correct color in highlighted choice
- Use $input-padding-y and $input-padding-x
- Use $input-bg as base color to darken for selected choice

### Fix cursor on disabled elements

Reproducible with Chrome 88 on macOS

#### Before
![image](https://user-images.githubusercontent.com/556268/105627855-8efca180-5e39-11eb-9dd0-9f4f9256aa73.png)

#### After
![image](https://user-images.githubusercontent.com/556268/105629238-1817d680-5e42-11eb-994e-92d2d608a308.png)

ttskch commented:
@tagliala Thanks for your contribution! I think your PR is great. But I've just pushed some commits, so could you please resolve the conflicts? 🙇

tagliala commented:
Hi @ttskch

I've rebased and I've added a fix for cursor in disabled elements

I've also add stylelint with (almost) the same configuration as bootstrap (I've included a regex to allo BEM-like selectors)

ttskch commented:
@tagliala Thanks a lot!
